tests/lib/eslint/eslint.js

```diff
@@ -437,6 +441,27 @@ describe("ESLint", () => {
 				assert.strictEqual(eslint.hasFlag("test_only"), true);
 			});
 
+			it("should return true if the flag is present and active with ESLINT_FLAGS", () => {
+				process.env.ESLINT_FLAGS = "test_only";
+				eslint = new ESLint({
+					cwd: getFixturePath(),
+					flags: ["test_only"],
+				});
+				assert.strictEqual(eslint.hasFlag("test_only"), true);
+			});
```

Comment:
```suggestion
			});
			
			it("should merge flags passed through API with flags passed through ESLINT_FLAGS", () => {
				process.env.ESLINT_FLAGS = "test_only";
				eslint = new ESLint({
					cwd: getFixturePath(),
					flags: ["test_only_2"],
				});
				assert.strictEqual(eslint.hasFlag("test_only"), true);
				assert.strictEqual(eslint.hasFlag("test_only_2"), true);
			});
```

It would be good to ensure the flags are merged. Currently, if we remove `...flags` from the merge function, no tests would fail.